json_rpc/clients/httpclient.nim

```diff
@@ -176,60 +171,36 @@ proc httpMethod*(client: RpcHttpClient): HttpMethod =
 proc httpMethod*(client: RpcHttpClient, m: HttpMethod) =
   client.options.httpMethod = m
 
-proc call*(client: RpcHttpClient, name: string,
-           params: JsonNode, httpMethod: HttpMethod): Future[Response] {.async.} =
+method call*(client: RpcHttpClient, name: string,
```

Comment:
Do I understand right that the client was originally reconnecting automatically each time in while loop after the request ended and that now this gets done properly on the request call itself?

Reply:
"the client was originally reconnecting automatically each time in while loop" - that was the intended behavior I think, but unfortunately it did not work. Sending multiple requests in a row was subjected to a race condition when the send proc tried to write to the socket, (temporarily) closed by the read loop. Since we're not using HTTP 1.1 persistent connection anyway I thought this simplification was a good idea. Funnily I did it before I found that HTTP proto is now deprecated altogether in web3, but oh well.